### PR TITLE
[DOC-1805] Document delegate APIs

### PR DESCRIPTION
**Pull request by KimberlyFields**
Finishing work that Katherine started in a different PR that was closed without being merged.

### CODE REVIEW

File: docs/platform/2_Delegates/delegate-reference/delegate-restful-resources/delegate-restful-resources-overview.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# sidebar_position: 2
+---
+
+Harness API provides the following RESTful resources for working with NextGen delegates. This topic introduces the resources and provides information about their requirements and possible workflows. For detailed usage information, see [Harness NextGen Platform API Reference](https://apidocs.harness.io/).
+
+| Resource | Operations |
+| :-- | :-- |
+| Delegate download resource<br />`download-delegates` | Download YAML files for Kubernetes and Docker delegates. |
+| Delegate group tags resource<br />`delegate-group-tags` | Add, retrieve, remove, delete, and list one or more descriptive tags from groups of one or more delegates. |
+| Delegate setup resource<br />`delegate-setup` | Perform setup tasks including generating a Helm values YAML file and deleting a delegate. |
```
Comment (AnupamIO) [inline docs/platform/2_Delegates/delegate-reference/delegate-restful-resources/delegate-restful-resources-overview.md:13]: This resource has more apis, please check: https://apidocs.harness.io/tag/Delegate-Setup-Resource

File: docs/platform/2_Delegates/delegate-reference/delegate-restful-resources/rest-resource-delegate-setup.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+title: Use REST resource delegate-setup
```
Comment (AnupamIO) [inline docs/platform/2_Delegates/delegate-reference/delegate-restful-resources/rest-resource-delegate-setup.md:2]: same as above, let's write description about newly added apis: https://apidocs.harness.io/tag/Delegate-Setup-Resource
Rest looks good.

**Comment by ravilach** [comment]
@KimberlyFields do we still need this? Was identified as a stale branch in GH.

**Comment by KimberlyFields** [comment]
Closing without merging because this content should be in the API docs [ ](https://apidocs.harness.io/)

**Pull request closed without merging**